### PR TITLE
Photoshop: add autocreators for review and flat image

### PR DESCRIPTION
**Pull request by kalisp**
## Changelog Description
Review and flatten image (produced when no instance of `image` family was created) were created somehow magically. This PR
introduces two new auto creators which allow artists to disable review or flatten image.

For all `image` instances `Review` flag was added to provide functionality to create separate review per `image` instance. Previously was possible only to have separate instance of `review` family.
Review is not enabled on `image` family by default. (Eg. follows original behavior)

Review auto creator is enabled by default as it was before.
Flatten image creator must be set in Settings in `project_settings/photoshop/create/AutoImageCreator`.

python_pubb2VjNBM

## Additional info
Both creators shouldn't affect publishing from Webpublisher.

## Testing notes:
1. open in Photoshop existing scene, do publish
2. two new instances should show up
3. play with disabling/enabling them, disabling/enabling review flags.

### CODE REVIEW

File: openpype/hosts/photoshop/lib.py [hunk 1]
```diff
@@ -7,28 +7,27 @@
...
-    default_variant = "Main"
-
+class PSAutoCreator(AutoCreator):
+    """Generic autocreator to extend."""
     def get_instance_attr_defs(self):
         return []
 
     def collect_instances(self):
         for instance_data in cache_and_get_instances(self):
+            self.log.debug("instance_data:: {}".format(instance_data))
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/lib.py:17]: This is a lot of the same logging happening in each auto creator.

File: openpype/hosts/photoshop/plugins/publish/collect_instances.py [hunk 2]
```diff
@@ -84,16 +88,28 @@ def process(self, context):
...
                              "Remove unwanted via Publisher")
 
-        if len(instance_names) == 0 and self.flatten_subset_template:
-            project_name = context.data["projectEntity"]["name"]
-            variants = get_project_settings(project_name).get(
+        # additional logic only for remote publishing from Webpublisher
+        if "remotepublish" not in pyblish.api.registered_targets():
+            return
+
+        project_name = get_current_project_name()
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/publish/collect_instances.py:95]: Shouldn't this happen in separated collector?

File: openpype/hosts/photoshop/plugins/publish/collect_instances.py [hunk 3]
```diff
@@ -84,16 +88,28 @@ def process(self, context):
...
+        proj_settings = get_project_settings(project_name)
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/publish/collect_instances.py:96]:
```suggestion
        proj_settings = context.data["project_settings"]
```

**Review by 64qam — APPROVED** [review]

File: openpype/hosts/photoshop/plugins/publish/collect_auto_workfile.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                return
+
+        stub = photoshop.stub()
+        stored_items = stub.get_layers_metadata()
+        for item in stored_items:
+            if item.get("creator_identifier") == family:
+                if not item.get("active"):
+                    self.log.debug("Workfile instance disabled")
+                    return
+
```
Comment (iLLiCiTiT) [inline openpype/hosts/photoshop/plugins/publish/collect_auto_workfile.py:53]: Is this needed when it's only in remotepublish target?
```suggestion
```

**Pull request merged**